File: 643-maximum-average-subarray-i/maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        long res = 0;
        for(int i = 0; i < k; i++){
            res += nums[i];
        }
        long tmp = res;
        for(int i = k; i < nums.length; i++){
            tmp = tmp + nums[i] - nums[i - k];
            if(tmp > res){
                res = tmp;
            }
        }
        float x = k;
        return (double)res / x;
    }
}

File: 643-maximum-average-subarray-i/maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        res = 0
        for i in range(k):
            res += nums[i]
        summ = res
        for i in range(k, len(nums)):
            summ -= nums[i - k]
            summ += nums[i]
            if res < summ:
                res = summ
        return res / k